pkg/minio/minio.go

```diff
@@ -0,0 +1,162 @@
+package minio
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+const (
+	MsgFailedInitiateClient string = "failed to initiate MinIO client: %w"
+	MsgFailedUpload         string = "failed to upload object: %w"
+	MsgFailedLoad           string = "failed to load object: %w"
+	MsgFailedGetInfo        string = "failed to get info for object: %w"
+	MsgFailedCreateBucket   string = "failed to create bucket: %w"
+)
+
+func init() {
+	fsMinIO := pflag.NewFlagSet("minio", pflag.ExitOnError)
+	fsMinIO.Bool("minio-enabled", false, "enable MinIO")
+	fsMinIO.String("minio-host", "minio.svc.local:9001", "MinIO host")
```

Comment:
I suppose, port `9001` is service UI port, correct Minio backend port is `9000` isn't it?